doc/decisions/DR020-Coding-style.md

```diff
@@ -0,0 +1,68 @@
+# DR020 Coding style
+
+- **Status:** Decided
+- **Driver:** @foundrytom
+- **Approver:** @felltech @carmenpinto @fnRaihanKibria
+- **Outcome:** OpenAssetIO will use the Google C++ Style Guide.
+
+## Background
+
+OpenAssetIO is a collaboratively maintained codebase. A consistent
+coding style can help simplify maintenance. As such, contributors need
+clear guidelines.
+
+## Options considered
+
+### Option 1 - Define a custom style guide
+
+Maintainers define a bespoke style guide that embodies their
+locally agreed preferences.
+
+#### Pros
+
+- Allows local consensus around what is preferable in the code base.
```

Comment:
[consider]

* Allows a style to be designed that best fits the project, e.g. to balance idiomatic differences in a multi-language project.
* Encourages constant adaptation and optimisation as the project and tooling changes over time.

Reply:
Added [https://github.com/OpenAssetIO/OpenAssetIO/pull/1096/commits/b4e33b673288fc113cbf740060a6cb9e739e1e2b](here).